filters/CMakeLists.txt

```diff
@@ -127,6 +127,13 @@ set(LIB_NAME "pcl_${SUBSYS_NAME}")
 include_directories("${CMAKE_CURRENT_SOURCE_DIR}/include")
 PCL_ADD_LIBRARY(${LIB_NAME} COMPONENT ${SUBSYS_NAME} SOURCES ${srcs} ${incs} ${impl_incs})
 target_link_libraries("${LIB_NAME}" pcl_common pcl_sample_consensus pcl_search pcl_kdtree pcl_octree)
+
+if(MSVC AND NOT (MSVC_VERSION LESS 1915))
+  # You must acknowledge that you understand MSVC resolved a byte alignment issue in this compiler
+  # We get this due to using Eigen objects and allocating those objects with make_shared
```

Comment:
Is this independent of the custom `make_shared` function in PCL?

Reply:
I think so, but it does sound strange that no other allocate_shared triggered this. Maybe there's none because the library code always goes for make_shared? Check https://github.com/PointCloudLibrary/pcl/pull/3750#issuecomment-599134630 and https://github.com/PointCloudLibrary/pcl/pull/3750#issuecomment-599152178 for reference.

Reply:
I missed them in the old PR.

>  I can't find eigen members in the neighborhood

`PointXYZ` uses the new function of `Eigen` to allocate, so there's the Eigen member in the neighborhood.

Reply:
Eh, I meant to say no Eigen member with a missing `PCL_MAKE_ALIGNED_OPERATOR_NEW`. Could it really be it's the only `pcl::make_shared` invocation in the codebase that triggers `std::allocate_shared`?

Reply:
Comment it out and see the magic... That's all I can say 😅

Reply:
I don't understand what "it" refers to

Reply:
> it's the only pcl::make_shared invocation in the codebase that triggers std::allocate_shared?

I was referring to this `it`

Reply:
<details>
<summary>This CMake configuration</summary>
<br>

```
BUILD_2d:BOOL=ON
BUILD_CUDA:BOOL=OFF
BUILD_GPU:BOOL=OFF
BUILD_apps:BOOL=ON
BUILD_apps_3d_rec_framework:BOOL=ON
BUILD_apps_cloud_composer:BOOL=ON
BUILD_apps_in_hand_scanner:BOOL=ON
BUILD_apps_modeler:BOOL=ON
BUILD_apps_point_cloud_editor:BOOL=ON
BUILD_common:BOOL=ON
BUILD_cuda_apps:BOOL=OFF
BUILD_cuda_common:BOOL=ON
BUILD_cuda_features:BOOL=ON
BUILD_cuda_io:BOOL=OFF
BUILD_cuda_sample_consensus:BOOL=ON
BUILD_cuda_segmentation:BOOL=ON
BUILD_examples:BOOL=ON
BUILD_features:BOOL=ON
BUILD_filters:BOOL=ON
BUILD_geometry:BOOL=ON
BUILD_global_tests:BOOL=ON
BUILD_gpu_containers:BOOL=ON
BUILD_gpu_features:BOOL=ON
BUILD_gpu_kinfu:BOOL=ON
BUILD_gpu_kinfu_large_scale:BOOL=ON
BUILD_gpu_octree:BOOL=ON
BUILD_gpu_people:BOOL=OFF
BUILD_gpu_segmentation:BOOL=ON
BUILD_gpu_surface:BOOL=OFF
BUILD_gpu_tracking:BOOL=OFF
BUILD_gpu_utils:BOOL=ON
BUILD_io:BOOL=ON
BUILD_kdtree:BOOL=ON
BUILD_keypoints:BOOL=ON
BUILD_ml:BOOL=ON
BUILD_octree:BOOL=ON
BUILD_outofcore:BOOL=ON
BUILD_people:BOOL=ON
BUILD_recognition:BOOL=ON
BUILD_registration:BOOL=ON
BUILD_sample_consensus:BOOL=ON
BUILD_search:BOOL=ON
BUILD_segmentation:BOOL=ON
BUILD_simulation:BOOL=ON
BUILD_stereo:BOOL=ON
BUILD_surface:BOOL=ON
BUILD_surface_on_nurbs:BOOL=ON
BUILD_tests_2d:BOOL=ON
BUILD_tests_common:BOOL=ON
BUILD_tests_features:BOOL=ON
BUILD_tests_filters:BOOL=ON
BUILD_tests_geometry:BOOL=ON
BUILD_tests_io:BOOL=ON
BUILD_tests_kdtree:BOOL=ON
BUILD_tests_keypoints:BOOL=ON
BUILD_tests_octree:BOOL=ON
BUILD_tests_outofcore:BOOL=ON
BUILD_tests_people:BOOL=ON
BUILD_tests_recognition:BOOL=ON
BUILD_tests_registration:BOOL=ON
BUILD_tests_sample_consensus:BOOL=ON
BUILD_tests_search:BOOL=ON
BUILD_tests_segmentation:BOOL=ON
BUILD_tests_surface:BOOL=ON
BUILD_tests_visualization:BOOL=ON
BUILD_tools:BOOL=ON
BUILD_tracking:BOOL=ON
BUILD_visualization:BOOL=ON
Boost_DATE_TIME_LIBRARY_DEBUG:FILEPATH=/usr/lib/x86_64-linux-gnu/libboost_date_time.so
Boost_DATE_TIME_LIBRARY_RELEASE:FILEPATH=/usr/lib/x86_64-linux-gnu/libboost_date_time.so
Boost_FILESYSTEM_LIBRARY_DEBUG:FILEPATH=/usr/lib/x86_64-linux-gnu/libboost_filesystem.so
Boost_FILESYSTEM_LIBRARY_RELEASE:FILEPATH=/usr/lib/x86_64-linux-gnu/libboost_filesystem.so
Boost_INCLUDE_DIR:PATH=/usr/include
Boost_IOSTREAMS_LIBRARY_DEBUG:FILEPATH=/usr/lib/x86_64-linux-gnu/libboost_iostreams.so
Boost_IOSTREAMS_LIBRARY_RELEASE:FILEPATH=/usr/lib/x86_64-linux-gnu/libboost_iostreams.so
Boost_LIBRARY_DIR_DEBUG:PATH=/usr/lib/x86_64-linux-gnu
Boost_LIBRARY_DIR_RELEASE:PATH=/usr/lib/x86_64-linux-gnu
Boost_MPI_LIBRARY_DEBUG:FILEPATH=Boost_MPI_LIBRARY_DEBUG-NOTFOUND
Boost_MPI_LIBRARY_RELEASE:FILEPATH=Boost_MPI_LIBRARY_RELEASE-NOTFOUND
Boost_REGEX_LIBRARY_DEBUG:FILEPATH=/usr/lib/x86_64-linux-gnu/libboost_regex.so
Boost_REGEX_LIBRARY_RELEASE:FILEPATH=/usr/lib/x86_64-linux-gnu/libboost_regex.so
Boost_SERIALIZATION_LIBRARY_DEBUG:FILEPATH=/usr/lib/x86_64-linux-gnu/libboost_serialization.so
Boost_SERIALIZATION_LIBRARY_RELEASE:FILEPATH=/usr/lib/x86_64-linux-gnu/libboost_serialization.so
Boost_SYSTEM_LIBRARY_DEBUG:FILEPATH=/usr/lib/x86_64-linux-gnu/libboost_system.so
Boost_SYSTEM_LIBRARY_RELEASE:FILEPATH=/usr/lib/x86_64-linux-gnu/libboost_system.so
CMAKE_AR:FILEPATH=/usr/bin/ar
CMAKE_BUILD_TYPE:STRING=RelWithDebInfo
CMAKE_COLOR_MAKEFILE:BOOL=ON
CMAKE_CONFIGURATION_TYPES:STRING=Debug;Release
CMAKE_CXX_COMPILER:FILEPATH=/usr/bin/c++
CMAKE_CXX_FLAGS:STRING=
CMAKE_CXX_FLAGS_DEBUG:STRING=-g
CMAKE_CXX_FLAGS_MAINTAINER:STRING=-pedantic -Wno-variadic-macros -Weffc++ -Wno-long-long
CMAKE_CXX_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG
CMAKE_CXX_FLAGS_RELEASE:STRING=-O3 -DNDEBUG
CMAKE_CXX_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
CMAKE_CXX_STANDARD:STRING=14
CMAKE_C_COMPILER:FILEPATH=/usr/bin/cc
CMAKE_C_FLAGS:STRING=
CMAKE_C_FLAGS_DEBUG:STRING=-g
CMAKE_C_FLAGS_MAINTAINER:STRING=-pedantic -Wno-variadic-macros -Weffc++ -Wno-long-long
CMAKE_C_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG
CMAKE_C_FLAGS_RELEASE:STRING=-O3 -DNDEBUG
CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
CMAKE_EXE_LINKER_FLAGS:STRING=
CMAKE_EXE_LINKER_FLAGS_DEBUG:STRING=
CMAKE_EXE_LINKER_FLAGS_MAINTAINER:STRING=-Wl,--warn-unresolved-symbols,--warn-once
CMAKE_EXE_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_EXE_LINKER_FLAGS_RELEASE:STRING=
CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_EXPORT_COMPILE_COMMANDS:BOOL=OFF
CMAKE_INSTALL_PREFIX:PATH=/usr/local
CMAKE_LINKER:FILEPATH=/usr/bin/ld
CMAKE_MAKE_PROGRAM:FILEPATH=/usr/bin/make
CMAKE_MODULE_LINKER_FLAGS:STRING=
CMAKE_MODULE_LINKER_FLAGS_DEBUG:STRING=
CMAKE_MODULE_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_MODULE_LINKER_FLAGS_RELEASE:STRING=
CMAKE_MODULE_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_MSVC_CODE_LINK_OPTIMIZATION:BOOL=ON
CMAKE_NM:FILEPATH=/usr/bin/nm
CMAKE_OBJCOPY:FILEPATH=/usr/bin/objcopy
CMAKE_OBJDUMP:FILEPATH=/usr/bin/objdump
CMAKE_RANLIB:FILEPATH=/usr/bin/ranlib
CMAKE_SHARED_LINKER_FLAGS:STRING=
CMAKE_SHARED_LINKER_FLAGS_DEBUG:STRING=
CMAKE_SHARED_LINKER_FLAGS_MAINTAINER:STRING=-Wl,--warn-unresolved-symbols,--warn-once
CMAKE_SHARED_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_SHARED_LINKER_FLAGS_RELEASE:STRING=
CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_SKIP_INSTALL_RPATH:BOOL=NO
CMAKE_SKIP_RPATH:BOOL=NO
CMAKE_STATIC_LINKER_FLAGS:STRING=
CMAKE_STATIC_LINKER_FLAGS_DEBUG:STRING=
CMAKE_STATIC_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_STATIC_LINKER_FLAGS_RELEASE:STRING=
CMAKE_STATIC_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_STRIP:FILEPATH=/usr/bin/strip
CMAKE_TIMING_VERBOSE:BOOL=OFF
CMAKE_VERBOSE_MAKEFILE:BOOL=FALSE
CPACK_SOURCE_TBZ2:BOOL=ON
CPACK_SOURCE_TGZ:BOOL=ON
CPACK_SOURCE_TXZ:BOOL=ON
CPACK_SOURCE_TZ:BOOL=ON
CPACK_SOURCE_ZIP:BOOL=OFF
CUDA_64_BIT_DEVICE_CODE:BOOL=ON
CUDA_ARCH_BIN:STRING=3.0 3.5 5.0 5.2 5.3 6.0 6.1 7.0
CUDA_ARCH_PTX:STRING=
CUDA_ATTACH_VS_BUILD_RULE_TO_CUDA_FILE:BOOL=ON
CUDA_BUILD_CUBIN:BOOL=OFF
CUDA_BUILD_EMULATION:BOOL=OFF
CUDA_CUDART_LIBRARY:FILEPATH=/usr/local/cuda/lib64/libcudart.so
CUDA_CUDA_LIBRARY:FILEPATH=CUDA_CUDA_LIBRARY-NOTFOUND
CUDA_GENERATED_OUTPUT_DIR:PATH=
CUDA_HOST_COMPILATION_CPP:BOOL=ON
CUDA_HOST_COMPILER:FILEPATH=/usr/bin/cc
CUDA_NVCC_EXECUTABLE:FILEPATH=/usr/local/cuda/bin/nvcc
CUDA_NVCC_FLAGS:STRING=
CUDA_NVCC_FLAGS_DEBUG:STRING=
CUDA_NVCC_FLAGS_MINSIZEREL:STRING=
CUDA_NVCC_FLAGS_RELEASE:STRING=
CUDA_NVCC_FLAGS_RELWITHDEBINFO:STRING=
CUDA_PROPAGATE_HOST_FLAGS:BOOL=ON
CUDA_SDK_ROOT_DIR:PATH=CUDA_SDK_ROOT_DIR-NOTFOUND
CUDA_SEPARABLE_COMPILATION:BOOL=OFF
CUDA_TARGET_CPU_ARCH:STRING=
CUDA_TOOLKIT_INCLUDE:PATH=/usr/local/cuda/include
CUDA_TOOLKIT_ROOT_DIR:PATH=/usr/local/cuda
CUDA_TOOLKIT_TARGET_DIR:PATH=/usr/local/cuda
CUDA_USE_STATIC_CUDA_RUNTIME:BOOL=ON
CUDA_VERBOSE_BUILD:BOOL=OFF
CUDA_VERSION:STRING=9.0
CUDA_cublas_LIBRARY:FILEPATH=/usr/local/cuda/lib64/libcublas.so
CUDA_cudart_static_LIBRARY:FILEPATH=/usr/local/cuda/lib64/libcudart_static.a
CUDA_cufft_LIBRARY:FILEPATH=/usr/local/cuda/lib64/libcufft.so
CUDA_cupti_LIBRARY:FILEPATH=/usr/local/cuda/extras/CUPTI/lib64/libcupti.so
CUDA_curand_LIBRARY:FILEPATH=/usr/local/cuda/lib64/libcurand.so
CUDA_cusolver_LIBRARY:FILEPATH=/usr/local/cuda/lib64/libcusolver.so
CUDA_cusparse_LIBRARY:FILEPATH=/usr/local/cuda/lib64/libcusparse.so
CUDA_nppc_LIBRARY:FILEPATH=/usr/local/cuda/lib64/libnppc.so
CUDA_nppi_LIBRARY:FILEPATH=CUDA_nppi_LIBRARY-NOTFOUND
CUDA_npps_LIBRARY:FILEPATH=/usr/local/cuda/lib64/libnpps.so
CUDA_rt_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/librt.so
ClangFormat_EXECUTABLE:FILEPATH=ClangFormat_EXECUTABLE-NOTFOUND
DAVIDSDK_INCLUDE_DIR:PATH=DAVIDSDK_INCLUDE_DIR-NOTFOUND
DAVIDSDK_LIBRARY:FILEPATH=DAVIDSDK_LIBRARY-NOTFOUND
DPKG_PROGRAM:FILEPATH=/usr/bin/dpkg
DSSDK_DIR:PATH=DSSDK_DIR-NOTFOUND
EIGEN_INCLUDE_DIR:PATH=/usr/include/eigen3
ENSENSO_INCLUDE_DIR:PATH=/opt/ensenso/development/c/include
ENSENSO_LIBRARY:FILEPATH=/usr/lib/libNxLib64.so
FLANN_INCLUDE_DIR:PATH=/usr/include
FLANN_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/libflann_cpp.so
FLANN_LIBRARY_DEBUG:FILEPATH=FLANN_LIBRARY_DEBUG-NOTFOUND
GLEW_INCLUDE_DIR:PATH=/usr/include
GLEW_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/libGLEW.so
GLUT_INCLUDE_DIR:PATH=GLUT_INCLUDE_DIR-NOTFOUND
GLUT_Xi_LIBRARY:FILEPATH=GLUT_Xi_LIBRARY-NOTFOUND
GLUT_Xmu_LIBRARY:FILEPATH=GLUT_Xmu_LIBRARY-NOTFOUND
GLUT_glut_LIBRARY:FILEPATH=GLUT_glut_LIBRARY-NOTFOUND
GTEST_INCLUDE_DIR:PATH=/usr/include
GTEST_SRC_DIR:PATH=/usr/src/gtest
LIBUSB_1_INCLUDE_DIR:PATH=/usr/include
LIBUSB_1_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/libusb-1.0.so
OPENGL_INCLUDE_DIR:PATH=/usr/include
OPENGL_gl_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/libGL.so
OPENGL_glu_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/libGLU.so
OPENGL_xmesa_INCLUDE_DIR:PATH=OPENGL_xmesa_INCLUDE_DIR-NOTFOUND
OPENNI2_INCLUDE_DIR:PATH=/usr/include/openni2
OPENNI2_LIBRARY:FILEPATH=/usr/lib/libOpenNI2.so
OPENNI_INCLUDE_DIR:PATH=/usr/include/ni
OPENNI_LIBRARY:FILEPATH=/usr/lib/libOpenNI.so
OpenMP_CXX_FLAGS:STRING=-fopenmp
OpenMP_C_FLAGS:STRING=-fopenmp
PCAP_INCLUDE_DIR:PATH=PCAP_INCLUDE_DIR-NOTFOUND
PCAP_LIBRARY:FILEPATH=PCAP_LIBRARY-NOTFOUND
PCL_BUILD_WITH_BOOST_DYNAMIC_LINKING_WIN32:BOOL=OFF
PCL_BUILD_WITH_FLANN_DYNAMIC_LINKING_WIN32:BOOL=OFF
PCL_BUILD_WITH_QHULL_DYNAMIC_LINKING_WIN32:BOOL=OFF
PCL_ENABLE_CCACHE:BOOL=OFF
PCL_ENABLE_SSE:BOOL=ON
PCL_IO_ENABLE_MAND_LOCKING:BOOL=ON
PCL_NO_PRECOMPILE:BOOL=OFF
PCL_ONLY_CORE_POINT_TYPES:BOOL=OFF
PCL_SHARED_LIBS:BOOL=ON
PCL_VERBOSITY_LEVEL:STRING=Info
PCL_WARNINGS_ARE_ERRORS:BOOL=OFF
PKG_CONFIG_EXECUTABLE:FILEPATH=/usr/bin/pkg-config
PNG_LIBRARY_DEBUG:FILEPATH=PNG_LIBRARY_DEBUG-NOTFOUND
PNG_LIBRARY_RELEASE:FILEPATH=/usr/lib/x86_64-linux-gnu/libpng.so
PNG_PNG_INCLUDE_DIR:PATH=/usr/include
QHULL_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/libqhull.so
QHULL_LIBRARY_DEBUG:FILEPATH=/usr/lib/x86_64-linux-gnu/libqhull.so
Qt5Concurrent_DIR:PATH=/usr/lib/x86_64-linux-gnu/cmake/Qt5Concurrent
Qt5Core_DIR:PATH=/usr/lib/x86_64-linux-gnu/cmake/Qt5Core
Qt5Gui_DIR:PATH=/usr/lib/x86_64-linux-gnu/cmake/Qt5Gui
Qt5OpenGL_DIR:PATH=/usr/lib/x86_64-linux-gnu/cmake/Qt5OpenGL
Qt5Widgets_DIR:PATH=/usr/lib/x86_64-linux-gnu/cmake/Qt5Widgets
Qt5_DIR:PATH=/usr/lib/x86_64-linux-gnu/cmake/Qt5
RSSDK_DIR:PATH=RSSDK_DIR-NOTFOUND
USB_10_INCLUDE_DIR:PATH=/usr/include
USB_10_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/libusb-1.0.so
USE_UMFPACK:BOOL=0
VTK_DIR:PATH=/usr/lib/cmake/vtk-6.2
WITH_CUDA:BOOL=ON
WITH_DAVIDSDK:BOOL=FALSE
WITH_DOCS:BOOL=OFF
WITH_DSSDK:BOOL=FALSE
WITH_ENSENSO:BOOL=ON
WITH_LIBUSB:BOOL=ON
WITH_OPENGL:BOOL=ON
WITH_OPENNI:BOOL=ON
WITH_OPENNI2:BOOL=ON
WITH_PCAP:BOOL=ON
WITH_PNG:BOOL=ON
WITH_QHULL:BOOL=ON
WITH_QT:BOOL=ON
WITH_RSSDK:BOOL=FALSE
WITH_RSSDK2:BOOL=ON
WITH_VTK:BOOL=ON
ZLIB_INCLUDE_DIR:PATH=/usr/include
ZLIB_LIBRARY_DEBUG:FILEPATH=ZLIB_LIBRARY_DEBUG-NOTFOUND
ZLIB_LIBRARY_RELEASE:FILEPATH=/usr/lib/x86_64-linux-gnu/libz.so
flann_DIR:PATH=flann_DIR-NOTFOUND
```
</details>

made with this diff
```
diff --git a/common/include/pcl/make_shared.h b/common/include/pcl/make_shared.h
index 30f6451..b2047ba 100644
--- a/common/include/pcl/make_shared.h
+++ b/common/include/pcl/make_shared.h
@@ -72,7 +72,7 @@ shared_ptr<T> make_shared(Args&&... args);
 template<typename T, typename ... Args>
 std::enable_if_t<has_custom_allocator<T>::value, shared_ptr<T>> make_shared(Args&&... args)
 {
-  return boost::allocate_shared<T>(Eigen::aligned_allocator<T>(), std::forward<Args> (args)...);
+//  return boost::allocate_shared<T>(Eigen::aligned_allocator<T>(), std::forward<Args> (args)...);
 }
 
 template<typename T, typename ... Args>
```

Compiles to 100%.

Reply:
> diff Compiles to 100%.

and 

> coming from a missing PCL_MAKE_ALIGNED_OPERATOR_NEW in some aligned class the code is instantiating with a make_shared instead of an `allocate_shared`

are contradictory. The call is going to an `allocate_shared`.

What happens if you comment L167 of `radius_outlier_removal.cpp`?

If `_ENABLE_EXTENDED_ALIGNED_STORAGE` is simply silencing the warning and not introducing any binary compatibility issues, I don't think we need to discuss anymore. Else, we might want to
* wait a bit past 1.10.1 release
* OR use `make_shared` normal with 1.10.1 and `allocate_shared` after release

Reply:
```
/pcl/filters/src/radius_outlier_removal.cpp: In member function 'virtual void pcl::RadiusOutlierRemoval<pcl::PCLPointCloud2>::applyFilter(std::vector<int>&)':
/pcl/filters/src/radius_outlier_removal.cpp:168:38: error: 'cloud' was not declared in this scope
   pcl::fromPCLPointCloud2 (*input_, *cloud);
                                      ^
```

with

```
diff --git a/filters/src/radius_outlier_removal.cpp b/filters/src/radius_outlier_removal.cpp
index 485bb80..73553be 100644
--- a/filters/src/radius_outlier_removal.cpp
+++ b/filters/src/radius_outlier_removal.cpp
@@ -164,7 +164,7 @@ pcl::RadiusOutlierRemoval<pcl::PCLPointCloud2>::applyFilter (std::vector<int> &i
     return;
   }
 
-  auto cloud = pcl::make_shared<pcl::PointCloud<pcl::PointXYZ>> ();
+//  auto cloud = pcl::make_shared<pcl::PointCloud<pcl::PointXYZ>> ();
   pcl::fromPCLPointCloud2 (*input_, *cloud);
 
   // Initialize the search class
```

Reply:
You need to comment the usage of cloud to create input (or create a dummy cloud)

Reply:
I honestly think it _is_ silencing the warning, since the message from CI was saying "Please define either (1) `_ENABLE_EXTENDED_ALIGNED_STORAGE` to _acknowledge that you understand this message and that you actually want a type with an extended alignment_, or (2) `_DISABLE_EXTENDED_ALIGNED_STORAGE` to silence this message and get the old non-conformant behavior." ([convenience link](https://github.com/PointCloudLibrary/pcl/pull/3750#issuecomment-599131027)). Maybe it should be `#define`d directly in place, however, since like this it's removing the warning from filters for all future work/changes.

Anyways with this diff
```
diff --git a/filters/src/radius_outlier_removal.cpp b/filters/src/radius_outlier_removal.cpp
index 485bb80..c801ee9 100644
--- a/filters/src/radius_outlier_removal.cpp
+++ b/filters/src/radius_outlier_removal.cpp
@@ -164,7 +164,8 @@ pcl::RadiusOutlierRemoval<pcl::PCLPointCloud2>::applyFilter (std::vector<int> &i
     return;
   }
 
-  auto cloud = pcl::make_shared<pcl::PointCloud<pcl::PointXYZ>> ();
+  pcl::PointCloud<pcl::PointXYZ>::Ptr cloud (new pcl::PointCloud<pcl::PointXYZ>);
+//  auto cloud = pcl::make_shared<pcl::PointCloud<pcl::PointXYZ>> ();
   pcl::fromPCLPointCloud2 (*input_, *cloud);
 
   // Initialize the search class
```

I get 100% compiling code.

Reply:
Since the issue belongs to `pcl::make_shared` for extended aligned storage, let's shift this to the CMake of `common` module and add a note in the function. Thanks for the investigation @aPonza 

I'm really baffled that this is the sole place where the compiler decides to make an issue. It should pop up in apps and visualization too at the very least. But the compiler knows better than me 😆